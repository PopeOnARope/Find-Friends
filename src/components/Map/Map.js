import { compose, withProps } from "recompose";
import { connect } from "react-redux";
import { toggleDetails } from "../../redux/reducers";
import glamorous from "glamorous";
import {
  withScriptjs,
  withGoogleMap,
  GoogleMap,
  Marker,
  InfoWindow
} from "react-google-maps";
import React from "react";

const WindowContent = glamorous.div(({ theme }) => ({
  display: "flex",
  flexDirection: "column",
  alignContent: "center",
  textAlign: "center",
  " h3": {
    fontSize: "1rem",
    color: `${theme.primaryDark}`
  },
  " p": {
    marginTop: "0px"
  },
  " img": {
    margin: "auto",
    width: "6rem"
  }
}));

const Map = props => (
  <GoogleMap
    defaultZoom={13}
    defaultCenter={{ lat: 32.797421, lng: -79.9412164 }}
  >
    {props.friends &&
      props.friends.map(
        ({ name, lng, lat, avatar, showDetails, info }, idx) => (
          <Marker
            position={{ lat, lng }}
            key={idx}
            onClick={e => props.toggleDetails({ idx })}
          >
            {showDetails && (
              <InfoWindow>
                <WindowContent>
                  <img src={avatar} alt={name} />
                  <h3>{name}</h3>
                  <p style={{ maxWidth: "150px" }}>{info}</p>
                </WindowContent>
              </InfoWindow>
            )}
          </Marker>
        )
      )}
  </GoogleMap>
);

const requiredProps = {
  googleMapURL:
    "https://maps.googleapis.com/maps/api/js?key=AIzaSyDpzkd_jDRkpqbXdetfATcqMJ417Mn2jqc&v=3.exp&libraries=geometry,drawing,places",
  loadingElement: <div style={{ height: `100%` }} />,
  containerElement: <div style={{ height: `400px` }} />,
  mapElement: <div style={{ height: `100%` }} />
};

const ComposedComponent = compose(
  withProps(requiredProps),
  withScriptjs,
  withGoogleMap
)(Map);

export default connect(state => state, { toggleDetails })(ComposedComponent);
