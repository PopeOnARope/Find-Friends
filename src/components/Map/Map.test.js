import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { ComposedComponent as Map } from "./Map";

configure({ adapter: new Adapter() });

const mockProps = {
  friends: [
    {
      avatar: "https://placekitten.com/g/25/26",
      name: "Jim",
      lat: 32.796421,
      lng: -79.9312164
    }
  ],
  toggleDetails: jest.fn()
};

it("Renders as expected", () => {
  const wrapper = shallow(<Map {...mockProps} />);
  expect(wrapper).toMatchSnapshot();
});
