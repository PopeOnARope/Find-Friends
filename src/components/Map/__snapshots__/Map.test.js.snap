// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withProps(withScriptjs(withGoogleMap(Map)))
    friends={
        Array [
            Object {
              "avatar": "https://placekitten.com/g/25/26",
              "lat": 32.796421,
              "lng": -79.9312164,
              "name": "Jim",
            },
          ]
    }
    toggleDetails={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "containerElement": <div
        style={
                Object {
                        "height": "400px",
                      }
        }
/>,
      "friends": Array [
        Object {
          "avatar": "https://placekitten.com/g/25/26",
          "lat": 32.796421,
          "lng": -79.9312164,
          "name": "Jim",
        },
      ],
      "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyDpzkd_jDRkpqbXdetfATcqMJ417Mn2jqc&v=3.exp&libraries=geometry,drawing,places",
      "loadingElement": <div
        style={
                Object {
                        "height": "100%",
                      }
        }
/>,
      "mapElement": <div
        style={
                Object {
                        "height": "100%",
                      }
        }
/>,
      "toggleDetails": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "containerElement": <div
          style={
                    Object {
                              "height": "400px",
                            }
          }
/>,
        "friends": Array [
          Object {
            "avatar": "https://placekitten.com/g/25/26",
            "lat": 32.796421,
            "lng": -79.9312164,
            "name": "Jim",
          },
        ],
        "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyDpzkd_jDRkpqbXdetfATcqMJ417Mn2jqc&v=3.exp&libraries=geometry,drawing,places",
        "loadingElement": <div
          style={
                    Object {
                              "height": "100%",
                            }
          }
/>,
        "mapElement": <div
          style={
                    Object {
                              "height": "100%",
                            }
          }
/>,
        "toggleDetails": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
